components/autofill/sql/create_shared_schema.sql

```diff
@@ -55,36 +55,36 @@ CREATE TABLE IF NOT EXISTS addresses_tombstones (
 -- whether the `cc_number` and/or other details should be encrypted or stored as plain text. Currently, we are storing
 -- them as plain text.
 CREATE TABLE IF NOT EXISTS credit_cards_data (
-    guid                TEXT NOT NULL PRIMARY KEY,
-    cc_name             TEXT NOT NULL, -- full name
-    cc_number           TEXT NOT NULL, -- TODO: consider storing this field as a hash
-    cc_exp_month        INTEGER,
-    cc_exp_year         INTEGER,
-    cc_type             TEXT NOT NULL,
-    -- cc_exp              TEXT NOT NULL, -- text format of the expiration date e.g. "[cc_exp_year]-[cc_exp_month]"
+    guid                    TEXT NOT NULL PRIMARY KEY,
+    billing_address_guid    TEXT NOT NULL, -- An optional GUID of an autofill address record which may or may not exist locally.
+    cc_name                 TEXT NOT NULL, -- full name
+    cc_number               TEXT NOT NULL, -- TODO: consider storing this field as a hash
+    cc_exp_month            INTEGER,
+    cc_exp_year             INTEGER,
+    cc_type                 TEXT NOT NULL,
 
-    time_created        INTEGER NOT NULL,
-    time_last_used      INTEGER,
-    time_last_modified  INTEGER NOT NULL,
-    times_used          INTEGER NOT NULL DEFAULT 0,
+    time_created            INTEGER NOT NULL,
+    time_last_used          INTEGER,
+    time_last_modified      INTEGER NOT NULL,
+    times_used              INTEGER NOT NULL DEFAULT 0,
 
     /* Same "sync change counter" strategy used by other components. */
     sync_change_counter INTEGER NOT NULL DEFAULT 1
 );
 
 CREATE TABLE IF NOT EXISTS credit_cards_mirror (
-    guid                TEXT NOT NULL PRIMARY KEY,
-    cc_name             TEXT NOT NULL, -- full name
-    cc_number           TEXT NOT NULL,
-    cc_exp_month        INTEGER,
-    cc_exp_year         INTEGER,
-    cc_type             TEXT NOT NULL,
-    -- cc_exp              TEXT NOT NULL, -- text format of the expiration date e.g. "[cc_exp_year]-[cc_exp_month]"
+    guid                    TEXT NOT NULL PRIMARY KEY,
+    billing_address_guid    TEXT NOT NULL, -- An optional GUID of an autofill address record which may or may not exist locally.
```

Comment:
I'm a bit confused by the intent here. `NOT NULL` implies it isn't optional. And the "may or may not exist" part also seems odd - does this mean this can hold a GUID that doesn't exist in the addresses table? If so, we should document better why that might happen - but if not (which I think we prefer), then a `FOREIGN KEY` constraint for this field probably makes sense.

Reply:
The main reason this is a WIP is because I'm honestly uncertain here. I pulled the comment from [desktop](https://searchfox.org/mozilla-central/source/browser/extensions/formautofill/FormAutofillStorage.jsm#61) and mean to review the desktop logic to understand why the billing address guid may or may not exist.

Reply:
After looking at desktop, the only reason I can find for `billing_address_guid` being null is a local address can't be found, as the comment states. While relevant for a JSON store, this shouldn't apply to our autofill component as we have a proper db that can have an FK constraint. I'm going to remove the comment and add the FK constraint.